edx/edx-enterprise
# ENT-1392 start date visibility for future courses

**Description:** Given that an enrollable course does not start until a future date, when a learner views the B2B landing page for that course, then the system should highlight the future start date so that it is clear to the learner.

**JIRA:** [1392](https://openedx.atlassian.net/browse/ENT-1392)

**Dependencies:** N/A

**Merge deadline:** N/A

**Installation instructions:** Install `edx-enterprise` from this branch `zub/ENT-1392-start-date-visibility` in `edx-platform`.

**Verification on Business Sandbox:**
* Login on Business Sandbox LMS with staff user.
* Old format for start date for courses started in past
https://business.sandbox.edx.org/enterprise/2b5a2956-7746-4fc9-9587-bc887ba45973/course/course-v1%3AedX%2BDemoX%2BDemo_Course/enroll/?new_enterprise_login=yes&tpa_hint=saml-pied-piper-testshib&catalog=cff09318-1358-4703-a277-bc2596480ed5
* New format for start date for courses in Future
https://business.sandbox.edx.org/enterprise/2b5a2956-7746-4fc9-9587-bc887ba45973/course/course-v1:SAPSuccessFactors+SAPSF101+2017_R1/enroll/?new_enterprise_login=yes&tpa_hint=saml-pied-piper-testshib&catalog=cff09318-1358-4703-a277-bc2596480ed5

**Testing instructions:**

1. Create an enterprise with a catalog
2. Add a course with course modes in the above enterprise catalog and publish it in course discovery service
3. Make sure that the course started in past, then run the `rebuild_index` command in course discovery service now verify that enterprise course enrollment page renders with default format for course start date
<img width="803" alt="screen shot 2019-01-03 at 6 13 55 pm" src="https://user-images.githubusercontent.com/5072991/50640414-a2085480-0f86-11e9-8a9a-2122714b24ae.png">

4. Now update start date of the course in future, then run the `rebuild_index` command in course discovery service now verify that enterprise course enrollment page renders with new format for start date to highlight the future start date and made it is clear to the learner
<img width="814" alt="screen shot 2019-01-03 at 6 11 16 pm" src="https://user-images.githubusercontent.com/5072991/50640551-062b1880-0f87-11e9-9016-b9f37c5dba9d.png">


**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
